### ISSUE TITLE
Law Details Page: "None" in last_updated, no Categories set

### ISSUE THREAD

**Issue by fin** [Forum-Informationsfreiheit/OffenesParlament#57]
It looks like I have reached an interesting state after a first scrape session:

![image](https://cloud.githubusercontent.com/assets/25808/14090897/33cbc2c4-f53d-11e5-9188-9f4d843cad84.png)
![image](https://cloud.githubusercontent.com/assets/25808/14090909/431beb78-f53d-11e5-86c4-e28b8c9db127.png)

No entry has a value in "Aktualisierung", but we shouldn't put "None" in there, IMHO.
Also, some don't have an "Art", even though they are the same parliament type (ME) as the others

**Comment by lyrixderaven** [comment]
Similar to the comment i just wrote for Ben's indexing issue, make sure the caching middleware in /offenesparlament/op_scraper/scraper/parlament/settings.py is not active any more. There's been some issues with it that lead to weird results, including having no timestamp set in the models (and subsequently in the index), as well as having the category (Art) not set. 

```
# Cache requests:
# DOWNLOADER_MIDDLEWARES = {
#     'scrapy.downloadermiddlewares.httpcache.HttpCacheMiddleware': 900,
# }
# HTTPCACHE_POLICY = 'scrapy.extensions.httpcache.DummyPolicy'
# HTTPCACHE_STORAGE = 'scrapy.extensions.httpcache.FilesystemCacheStorage'
# HTTPCACHE_ENABLED = True
# HTTPCACHE_DIR = '/tmp/scrapy-cache'
```

To verify these issues, please also compare the actual values in the database with the values in the index, that way we can pinpoint if the index is correct (meaning, the values really aren't in the db) or if the indexing process has some issues.

**Comment by fin** [comment]
since the last rebuild_index i've already re-scraped things, so the middlewares may be the issue - or not.
currently re-indexing in my VM, but it seems to take forever (:

**Comment by fin** [comment]
phew, a rebuild_index didn't fix this.

@benfreu, i could check out how to make sure we don't show "None" in the timestamp column, if you don't have other plans.

**Comment by lyrixderaven** [comment]
Please check what I suggested, is the TS _really_ None in the postgresdb or is it an indexing issue? Try re-scraping the persons and look at the debug output for the Person in question to see if there were any response timeout problems with the site. If the db is correct but the index is not, for some reason, check out the debug_trigger_changes.py script, it will show you how you can reindex a single object in the django shell, try that and see if it changes anything.

**Comment by lyrixderaven** [comment]
Alright, I checked it out, it's a frontend problem. Ben isn't using the timestamp for the Letze Aktualisierung field (for some reason?) but the following annotation:

```
    laws = Law.objects \
        .filter(legislative_period=llp) \
        .annotate(last_update=Max('steps__date')) \
        .order_by('-last_update') \
        .select_related('category')[:100]
```

Now, I don't know about why we are doing this, so I'm not touching it, but just as a hint: if I change the lawlist.html template like this:

```
<td class="lawlist_date">{{ law.ts }}</td>
```

I'm getting the actual last date that law was updated.

Furthermore about the missing category: that seems to be an actual scraper issue where the category isn't found. This we might have to accept for now, it's far too late to start meddling with the extractors, unless there's an obvious pattern to it (i.e., all laws with bla or foo don't get categories).

I'm assigning this to Ben in the meantime for the timestamp fix.

**Comment by lyrixderaven** [comment]
Alright, I checked it out, it's a frontend problem. Ben isn't using the timestamp for the Letze Aktualisierung field (for some reason?) but the following annotation:

```
    laws = Law.objects \
        .filter(legislative_period=llp) \
        .annotate(last_update=Max('steps__date')) \
        .order_by('-last_update') \
        .select_related('category')[:100]
```

Now, I don't know about why we are doing this, so I'm not touching it, but just as a hint: if I change the lawlist.html template like this:

```
<td class="lawlist_date">{{ law.ts }}</td>
```

I'm getting the actual last date that law was updated.

Furthermore about the missing category: that seems to be an actual scraper issue where the category isn't found. This we might have to accept for now, it's far too late to start meddling with the extractors, unless there's an obvious pattern to it (i.e., all laws with bla or foo don't get categories).

I'm assigning this to Ben in the meantime for the timestamp fix.
